### ISSUE TITLE
Sorted sets zrange command broken?

### ISSUE THREAD

**Issue by robacarp** [jgaskins/redis#49]
Thank you again for your quick help on #47 . After updating to 0.8.1 I started seeing breakages in some logic in [some reasonably aged code](https://github.com/mosquito-cr/mosquito/blob/master/src/mosquito/redis_backend.cr#L123-L130) which deals with sorted sets. Previously I was using 0.7.0.

It's possible I'm doing something wrong and I was using an API that shouldn't have worked and has since been fixed, but I didn't see anything in the commit history. 

I'm able to reproduce with this code sample:

```crystal
require "redis"

redis = Redis::Client.new(URI.parse("redis://localhost:6379"))

key = "fizz:buzz"
redis.zadd(key, 0, "foo")
redis.zadd(key, 0, "bar")
redis.zrange(key, 0, -1)
```

Stack trace:

```
Unhandled exception: ERR value is not an integer or out of range (Redis::Error)
  from lib/redis/src/connection.cr:421:9 in 'read'
  from lib/redis/src/connection.cr:357:18 in 'run'
  from lib/redis/src/connection.cr:351:5 in 'run'
  from lib/redis/src/client.cr:71:24 in 'run'
  from lib/redis/src/commands/sorted_set.cr:21:5 in 'zrange'
  from zset_test.cr:8:1 in '__crystal_main'
```

Here's what I think should be happening:

```redis
> flushdb
OK
> zadd fizz:buzz 0 foo
(integer) 1
> zadd fizz:buzz 0 bar
(integer) 1
> zrange fizz:buzz 0 -1
1) "bar"
2) "foo"
```

**Comment by jgaskins** [comment]
Oof, this is a missing `.to_s` in the command being sent to the Redis server. The server only accepts arrays of strings as commands, but the command is might be as `["zrange", "your-key", 0, -1]`. Basically, [these methods](https://github.com/jgaskins/redis/blob/06b7aa4164b06c588d86ed4ba89c042b118943cc/src/writer.cr#L24-L32) were a mistake to add. I didn't know it at the time, though.

As a stopgap, you can stringify the numbers passed to the `zrange` call, but I'll add the `.to_s` and fix the `Writer` so it can't encode things that aren't arrays of strings.

**Comment by jgaskins** [comment]
j/k That's not the problem — I misread the error message. It's that they're upcasted to `Float64`. For some reason I had it in my head that scores could be floating-point, but they're supposed to be integers.

**Comment by jgaskins** [comment]
Fixed in https://github.com/jgaskins/redis/commit/fbd617f70388eedb4c842b190d6a30fa2871c00d and released v0.9.0

**Comment by robacarp** [comment]
Thank you @jgaskins I appreciate you resolving this so quickly!
